File: ch07-control-structures/src/main/scala/NoBreakContinue.java

class NoBreakContinue {
    public static void main(String[] args) {
        
        int i = 0;
        boolean foundIt = false;
        while (i < args.length) {
            // System.out.println(args[i]);
            if (args[i].startsWith("-")) {
                i = i + 1;
                continue;
            }
            if (args[i].endsWith(".scala")) {
                foundIt = true;
                break;
            }
            i = i + 1;
        }
        System.out.println(i);
        System.out.println(args[i]);
    }
}

File: ch07-control-structures/src/main/scala/no-break-continue.scala
var i = -1

var foundIt = false
while (i < args.length && !foundIt) {
    i = i + 1
  if (!args(i).startsWith("-")) {
    if (args(i).endsWith(".scala"))
      foundIt = true
  }
}
println(args(i))

println
def searchFrom(i: Int): Int =
  if (i >= args.length) -1
  else if (args(i).startsWith("-")) searchFrom(i + 1)
  else if (args(i).endsWith(".scala")) i
  else searchFrom(i + 1)

val j = searchFrom(0)
println(args(j))
